collectors/diskspace.plugin/plugin_diskspace.c

```diff
@@ -671,7 +674,10 @@ void *diskspace_main(void *ptr) {
 
         struct mountinfo *mi;
         for(mi = disk_mountinfo_root; mi; mi = mi->next) {
-            if(unlikely(mi->flags & (MOUNTINFO_IS_DUMMY | MOUNTINFO_IS_BIND)))
```

Comment:
I suggest:
 - Do not change this part. I see that we intended to filter out all bind mounts (but fail to detect/set MOUNTINFO_IS_BIND).
 - Do not add "ignore duplicate devices and binds" because my understanding is that we never want to collect space metrics for bind mounts.

Reply:
If that is the case this PR is wrong.

Reply:
On Linux at least, the order of mount entries is based solely on the kernel’s internal mount table, and that, in turn, is currently ordered solely based on when the filesystems were mounted relative to each other (that is, filesystems that were mounted later come later in the table), though I’m not sure if that sort order is actually guaranteed.

Reply:
I can't find a better solution that this one, so I suggest we follow this approach. If the original mount ends up coming after the bind mount for whatever reason (e.g. maybe if it gets remounted with different options? I don't know, it shouldn't happen easily), then that's still not a major problem in my view, as long as metrics and alerts still get reported. 

The only additional I would suggest would be to also check for the shortest relative path to the root of the mounted file system, as this answer suggests:
> The following Python script lists all bind mounts. It assumes that the oldest mount point with the shortest relative path to the root of the mounted file system is the original mount.

https://unix.stackexchange.com/questions/18048/list-only-bind-mounts/346444#346444

Reply:
@MrZammler if we go with the assumption that

> he binded mounts come after the "real" one
will never happen

Reply:
@ilyam8 Done!

@Dim-P I'll check for that, thanks.

Reply:
We can also check with a usb stick and bind mounts under the usb stick (so our "root" will be something else that `/`).

Reply:
For some reason, there are no diskspace and inode charts for my USB stick which is mounted as `/run/media/dimitris/2849-75E9`, even though it shows up as a section in `netdata.conf`, and I've explicitly set 
```
[plugin:proc:diskspace:/run/media/dimitris/2849-75E9]
        space usage = yes
        inodes usage = yes
```

It doesn't seem to be related to this PR.